src/praisonai-bot/praisonai_bot/gateway/diagnostics.py

```
@@ -0,0 +1,238 @@
+
+def _now_stamp() -> str:
+    return _dt.datetime.now().strftime("%Y%m%d-%H%M%S")
+
+
+def _redact(text: str) -> str:
+    """Pass text through the shared secret-redaction registry (best-effort)."""
+    try:
+        from praisonaiagents.secrets import redact_secrets
+
+        return redact_secrets(text)
+    except Exception:  # pragma: no cover — never let redaction failure block
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)